samples/logging/prometheus-grafana-opentelemetry/sample.md

```diff
@@ -0,0 +1,134 @@
+```
+
+Custom observers need to be registered for the metric probes provided via `NServiceBus.Metrics`. This is all setup in the `PrometheusFeature`
+
+
+The registered observers convert `NServiceBus.Metric` *Signals* to Prometheus *Counters* and `NServiceBus.Metric` *Durations* to Prometheus *Summaries*.  Additionally, labels are added that identify the endpoint, the endpoint queue and more within Prometheus. With these labels, it is possible to filter and group metric values. 
+
+WARNING: Labels should be chosen thoughtfully since each unique combination of key-value label pairs represents a new time series which can dramatically increase the amount of data stored. The labels used here are for demonstration purpose only.
+
+During the registration the following steps are required:
+
+ * Map metric names
+ * Register observer callbacks
+ * Create summaries and counters with corresponding labels
+ * Invoke the summaries and counters in the observer callback
+
+## Docker stack
+
+Prometheus needs to be configured to get the metric data from the endpoint. Grafana also needs to be configured to get the data from Promethus and visualize it as graphs.
+
+To run the docker stack, run `docker-compse up -d` in the directory where the `docker-compose.yml` file is located.
+
+### Configuring Prometheus
+
+Copy the following files into the mapped volumes of the Prometheus and Grafana.
+
+ * `prometheus_ds.yml` should be copied to `./grafana/provisioning/datasources` folder
+ * `prometheus.yml` should be copied to `./prometheus` folder
+
+Open `prometheus.yml` and update the target IP address. This should be the address of the machine running the sample, and docker should be able to reach it. 
+
+```yml
+  - targets:
+    - '192.168.0.10:9184'
+```
+
+### Show a graph
+
+Start Prometheus by running the docker stack. NServiceBus pushes events for *success, failure, and fetched*. These events need to be converted to rates by a query:
+
+```
+avg(rate(messaging_successes[5m]))
+```
+
+![Prometheus graphs based on query](example-prometheus-graph.png)
+
+## Grafana
+
+Grafana needs to be installed and configured to display the data available in Prometheus. For more information how to install Grafana refer to the [Installation Guide](https://docs.grafana.org/installation).
+
+#### Dashboard
+
+To graph the Prometheus rule  `nservicebus_failure_total:avg_rate5m` the following steps have to be performed:
+
+ * Add a new dashboard
+ * Add a graph
+ * Click its title to edit
+ * Click the Metric tab
+
+<!-- ![Grafana metric using Prometheus as datasource](grafana-metric.png) -->
+
+### Dashboard
+
+![Grafana dashboard with NServiceBus OpenTelemetry metrics](example-grafana-dashboard.png)
+
+The sample included an [export of the grafana dashboard](grafana-endpoints-dashboard.json), this can be [imported](https://docs.grafana.org/reference/export_import/) as a reference.
```

Comment:
```suggestion
The sample includes an [export of the grafana dashboard](grafana-endpoints-dashboard.json), this can be [imported](https://docs.grafana.org/reference/export_import/) as a reference.
```

Reply:
Updated the wording